OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -27,6 +27,13 @@ public class ProvidesPrerequisiteInfo : ITraitInfo
 
 		[Desc("Should it recheck everything when it is captured?")]
 		public readonly bool ResetOnOwnerChange = false;
+
+		[Desc("Does the prerequisite need to be triggered before use?")]
+		public readonly bool NeedsActivation = false;
```

Comment:
This appears to be duplicating our existing upgrades logic.  You can instead make this implement `UpgradableTrait`, and then override the `UpgradeLevelChanged` method it exposes to replace ActivatePrerequisite.